### PR TITLE
[SPARK-29521][SQL] LOAD DATA INTO TABLE should look up catalog/table like v2 commands

### PR DESCRIPTION
**Pull request by viirya**
### What changes were proposed in this pull request?

Add LoadDataStatement and make LOAD DATA INTO TABLE go through the same catalog/table resolution framework of v2 commands.

### Why are the changes needed?

It's important to make all the commands have the same table resolution behavior, to avoid confusing end-users. e.g.

```
USE my_catalog
DESC t // success and describe the table t from my_catalog
LOAD DATA INPATH 'filepath'  INTO TABLE t // report table not found as there is no table t in the session catalog
```

### Does this PR introduce any user-facing change?

yes. When running LOAD DATA INTO TABLE, Spark fails the command if the current catalog is set to a v2 catalog, or the table name specified a v2 catalog.

### How was this patch tested?

Unit tests.

### CODE REVIEW

**Comment by viirya** [comment]
retest this please

File: sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala [hunk 1]
```diff
@@ -1210,6 +1210,38 @@ class DataSourceV2SQLSuite
...
+    val t = "testcat.ns1.ns2.tbl"
+    withTable(t) {
+      sql(
+        s"""
+           |CREATE TABLE $t (id bigint, data string)
+           |USING foo
+           |PARTITIONED BY (id)
+         """.stripMargin)
+
+      val e1 = intercept[AnalysisException] {
```
Comment (imback82) [inline sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala]: Refactor this a bit?
```Scala
def validate(sqlStr: String): Unit = {
  val e = intercept[AnalysisException] {
    sql(sqlStr)
  }
  assert(e.message.contains("LOAD DATA is only supported with v1 tables"))
}

validate(s"LOAD DATA INPATH 'filepath' INTO TABLE $t")
validate(s"LOAD DATA LOCAL INPATH 'filepath' INTO TABLE $t")
...
```

Or how about creating a function that [SPARK-29481](https://issues.apache.org/jira/browse/SPARK-29481) can reuse?

Comment (imback82) [reply sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala]: I actually created the function that can be reused here: #26183. Please let me know what you think. Thanks!

Comment (viirya) [reply sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala]: It looks good. Then I won't create another here, I can rebase this after #26183 is merged.

**Review by imback82 — APPROVED** [review]
LGTM

**Comment by viirya** [comment]
cc @cloud-fan

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Pull request closed without merging**